### ISSUE TITLE
dr_param now explicitly depends on catch2 and estd

### ISSUE THREAD

**Issue by wilson-ko** [fizyr/dr_param#1]
Hi,

Is it possible to remove the explicit dependency on catch2 here?

https://github.com/fizyr/dr_param/blob/master/package.xml#L16

We are working on our CI system now and this makes it difficult to manage our dependencies.. But the main reason why I ask this is because there is in my opinion no need for this explicit dependency, as shown in `estd`. Since `catch` is header only, I think it may be better for you to just provide the `hpp` file so that you don't force the user to install `catch` themselves:

https://github.com/fizyr/estd/blob/master/test/catch.hpp

The same goes for the explicit dependency on `estd` actually, since it's all header only, would it be an option to provide it as a submodule? The only downside to this is I guess that you would get multiple copies on your machine if multiple things depend on `estd`, so I'm also not very sure if this is a viable option. I do think that it would be good for `catch`, since that's only one single header file.

**Comment by hgaiser** [comment]
Having it as a submodule isn't really an option. It won't be included in the release package as far as I know and it is more tricky since every package that depends on estd would need to be updated when estd updates. The same holds for catch2 by the way, but that is less likely to cause issues. I would prefer that estd remains an explicit dependency. catch2 could be made into a `test_depend`, if that helps?

Why can't your CI handle header only dependencies? Seems like you have some work to do :p

**Comment by wilson-ko** [comment]
I think that the problem is that we are now using `rosdep`, which is rather limited in handling dependencies.. We are looking into using a different dependency manager but regardless, I think that since this is a ROS package that we need to make this work the ROS way too.

A problem now for example is that there are dependencies specified in the `package.xml` but it is assumed that it's somehow installed by the user.. So if it's not installed on the system not only will it not compile, but it's not possible for the user to know where you can get it from. I think the ROS way is using `rosdep install`, for which you'd need to add catch2 and estd to rosdistro:

Basically make a PR here https://github.com/ros/rosdistro/blob/master/melodic/distribution.yaml

Anyway for us the solution is to use a different dependency manager but you may want to look into making this work with `rosdep` since this is a ROS package after all. Also I personally don't have experience with this so take my comment with a grain of salt, but perhaps my colleague will explain more in this thread if I said anything wrong.

**Comment by hgaiser** [comment]
Right, but what is the alternative? Not put any dependency in package.xml that isn't in the rosdep database? That isn't really an option.

If you must use rosdep, I would say the issue is that rosdep does not allow to extend the list of packages in the command. My advice: ditch rosdep :p

**Comment by wilson-ko** [comment]
No I think putting the dependency in package.xml is good, but then maybe it would be good to update the rosdep database by making a PR here:

https://github.com/ros/rosdistro/blob/master/melodic/distribution.yaml

I basically agree with your advice which is what we're going to do (so it's up to you whether you want to make this work with rosdep or not; we don't really care anymore :P), but I can imagine that it may help some people who do want to use `rosdep`.

**Comment by hgaiser** [comment]
If it is in rosdep it has to be built according to the Ros [buildfarm](http://wiki.ros.org/buildfarm) configuration, has to be tested on different versions, built for different distros, etc. We did that a while ago for our calibration package, we won't be doing it again :p

In that case I'll close this issue.
